sdk/vision/azure-ai-vision-imageanalysis/setup.py

```diff
@@ -0,0 +1,72 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) Python Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
+# coding: utf-8
+
+import os
+import re
+from setuptools import setup, find_packages
+
+
+PACKAGE_NAME = "azure-ai-vision-imageanalysis"
+PACKAGE_PPRINT_NAME = "Azure Ai Vision Imageanalysis"
+
+# a-b-c => a/b/c
+package_folder_path = PACKAGE_NAME.replace("-", "/")
+
+# Version extraction inspired from 'requests'
+with open(os.path.join(package_folder_path, "_version.py"), "r") as fd:
+    version = re.search(r'^VERSION\s*=\s*[\'"]([^\'"]*)[\'"]', fd.read(), re.MULTILINE).group(1)
+
+if not version:
+    raise RuntimeError("Cannot find version information")
+
+
+setup(
+    name=PACKAGE_NAME,
+    version=version,
+    description="Microsoft {} Client Library for Python".format(PACKAGE_PPRINT_NAME),
+    long_description=open("README.md", "r").read(),
+    long_description_content_type="text/markdown",
+    license="MIT License",
+    author="Microsoft Corporation",
+    author_email="azpysdkhelp@microsoft.com",
+    url="https://github.com/Azure/azure-sdk-for-python/tree/main/sdk",
+    keywords="azure, azure sdk",
+    classifiers=[
+        "Development Status :: 4 - Beta",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3 :: Only",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.7",
```

Comment:
Let's remove 3.7 and add 3.12. :)

Reply:
Hi @xiangyan99 thank you for your comment! I can certainly do this change, however this file `setup.py` is auto-generated from the TypeSpec emitter. So it sounds like there is a work item on the Python emitter to remove support for 3.7 and add support for 3.12 (and 3.13?). Should I open a GitHub issue in azure-rest-api-specs repo? I don't mind doing the change here, just want to be sure that when the SDK is re-generated my change may be reverted.

Reply:
Good point. Yes. Let's open an issue to track it.

And FYI, this is our general python version support policy: https://github.com/Azure/azure-sdk-for-python/wiki/Azure-SDKs-Python-version-support-policy

Reply:
I updated setup.py to remove 3.7 and add 3.12 from the supported version. Also set minimum version to 3.8. 
I opened this GitHub issue to ask for the auto-generated setup.py to be updated accordingly: https://github.com/Azure/azure-rest-api-specs/issues/27170
Thanks @xiangyan99 !